### PR TITLE
Lint files in `lib/controller` directory.

### PR DESCRIPTION
**Pull request by tzmanics**
Ran a few tests but will test completely after review.

### CODE REVIEW

File: lib/controllers/addOn.js [hunk 1]
```diff
@@ -1,67 +1,54 @@
...
 AddOn.prototype.getById = function (id, callback) {
-  librarian.addOns.getById(id, function (err, addon) {
-    if (err) {
-      return callback(Errors.getMessage(err));
-    }
-
-    callback(null, addon);
-  });
+  Librarian.addOns.getById(
+    id, function (err, addon) {
```
Comment (franvarney) [inline lib/controllers/addOn.js]: I find this being split up hard to read..

File: lib/controllers/addOn.js [hunk 2]
```diff
@@ -1,67 +1,54 @@
...
-      return callback(Errors.getMessage(err));
-    }
-
-    callback(null);
-  });
+AddOn.prototype.deprovision = function (projectId, modulusId, callback) {
+  Librarian.addOns.deprovision(
+    projectId, modulusId, UserConfig.data.apiKey, function (err) {
+      if (err) return callback(Errors.getMessage(err));
+      callback(null);
```
Comment (franvarney) [inline lib/controllers/addOn.js]: Could remove the `null`.

File: lib/controllers/project.js [hunk 3]
```diff
@@ -21,779 +22,720 @@
...
-var packer = require('zip-stream');
-var through = require('through');
-var split = require('split');
-
-http.globalAgent.maxSockets = Number.MAX_VALUE;
-https.globalAgent.maxSockets = Number.MAX_VALUE;
-
-//-----------------------------------------------------------------------------
-var Project = function() {
+const Util = require('util');
```
Comment (franvarney) [inline lib/controllers/project.js]: Alphabetical ordering would make this much easier to read.

File: lib/controllers/project.js [hunk 4]
```diff
@@ -21,779 +22,720 @@
...
-      } else {
-        setTimeout(checkResult, 1000);
-      }
-    };
+// ----------------------------------------------------------------------------
+Project.prototype.delete = function (projectId, callback) {
+  Librarian.project.delete(
+    projectId, UserConfig.data.apiKey, function (err, result) {
+      var dbar, checkResult;
+      if (err) return callback(err, null);
```
Comment (franvarney) [inline lib/controllers/project.js]: Could remove this `null` and the one at 103. Doesn't affect functionality so, IMO, doesn't need to wait until a refactor.

File: lib/controllers/addOn.js (hunk 1 above)
Comment (jackboberg) [inline lib/controllers/addOn.js]: ^ Probably for consistency with the rest of the long line calls below. It's ugly, but the real culprit here is `librarian` exposing functions that take too many parameters.

File: lib/controllers/user.js [hunk 5]
```diff
@@ -34,68 +39,65 @@ var User = function() {
...
+User.prototype.authenticate = function (login, password, callback) {
+  var hashPass = Librarian.Util.createHash(password);
+  Librarian.user.authenticate(login, hashPass, callback);
 };
 
-User.prototype.authenticateGithub = function(login, password, callback) {
-  var token = null
-    , user = new Buffer(util.format('%s:%s', login, password), 'ascii').toString('base64');
+User.prototype.authenticateGithub = function (login, password, callback) {
+  var token = null;
```
Comment (franvarney) [inline lib/controllers/user.js]: Could remove the `null` from here and 128

File: lib/controllers/project.js [hunk 6]
```diff
@@ -21,779 +22,720 @@
...
+const Ignore = require('fstream-ignore');
+const Demeteorizer = require('demeteorizer');
+const Packer = require('zip-stream');
+const Through = require('through');
+const Split = require('split');
+
+const Modulus = require('../modulus');
+const Librarian = require('../common/api').librarian;
+const UserConfig = require('../common/api').userConfig;
+const ProgressBar = require('progress');
```
Comment (jackboberg) [inline lib/controllers/project.js]: should be grouped above

File: lib/controllers/project.js [hunk 7]
```diff
@@ -21,779 +22,720 @@
...
-        switch(type) {
-          case Project.types.NODEJS:
-            cb(null, dir);
-            break;
-          case Project.types.METEOR:
-            includeModules = true;
-            modulus.io.print('Meteor project detected...');
+      switch (type) {
+        case Project.types.NODEJS:
+          return cb(null, dir);
```
Comment (jackboberg) [inline lib/controllers/project.js]: I think these should keep the `break`, instead of some using `return`

**Comment by franvarney** [comment]
:+1:

File: lib/controllers/project.js [hunk 8]
```diff
@@ -21,779 +22,720 @@
...
+    function (path, zipFile, cb) {
+      projectController._deploy(
+        projectId, zipFile, registry, withTests, function (err, domain) {
+          FS.unlinkSync(zipFile);
 
+          //  Removes the .demeteorized folder
+          if (projectType === Project.types.METEOR) FsTools.removeSync(path);
           if (err) {
-            console.log(); //provides a newline
+            print('\n');
```
Comment (jackboberg) [inline lib/controllers/project.js]: in other locations I have seen `print(' ')` making me think each call injects a new-line, will this result in a double new-line?

File: lib/controllers/project.js [hunk 9]
```diff
@@ -21,779 +22,720 @@
...
-          logLength = log.length;
-          break;
-        }
+  client.on('end', function () {
+    var deployLogs, getDeployLogs, projectStatus, logLength = 0;
+    getDeployLogs = function (cb) {
+      Librarian.project.getDeployLogs(
+        projectId, UserConfig.data.apiKey, function (err, result) {
+          var key, log;
+          if (err) return callback(Errors.getMessage(err));
```
Comment (jackboberg) [inline lib/controllers/project.js]: added error handling may be better suited to refactor work to come

File: lib/controllers/project.js [hunk 10]
```diff
@@ -21,779 +22,720 @@
...
-              return callback('There was a problem deploying your project.');
+                break;
+              case Project.status.running:
+                if (status !== Project.status.uploading) {
+                  getDeployLogs(function () {
+                    print(' ');
+                    callback(null, proj.domain);
+                  });
+
+                  return callback();
```
Comment (jackboberg) [inline lib/controllers/project.js]: added `callback()` to return?

File: lib/controllers/project.js [hunk 11]
```diff
@@ -21,779 +22,720 @@
...
       }
     }
-    var dir = '';
-    if (path.dirname(outLocation) === '.'){
-      dir = process.cwd();
-    } else {
-      dir = path.dirname(path.resolve(outLocation));
-    }
-    outFile = outFile.replace(/ /g, '_').replace(/:/g,'.').replace(/[[\]{}@~`<>()*+=%&!?,\\^$|#\s]/g,'');
+    dir = '';
```
Comment (jackboberg) [inline lib/controllers/project.js]: I don't think this assignment is needed anymore

File: lib/controllers/project.js [hunk 12]
```diff
@@ -21,779 +22,720 @@
...
-        res.pipe(process.stdout);
-      }
-      res.on('error', function () { done(); });
-      res.on('close', function () { done(); });
-      res.on('end', function () { done(); });
+    req = h.get(endpoint, function (res) {
+      if (servos.length > 1) res.pipe(Split()).pipe(tr).pipe(process.stdout);
+      else res.pipe(process.stdout);
+      res.on('error', function () {
+        done();
```
Comment (jackboberg) [inline lib/controllers/project.js]: `done` doesn't care about arguments, so all of these could just be: `res.on(EVENT, done)`

...maybe for refactor branch

File: lib/controllers/user.js [hunk 13]
```diff
@@ -34,68 +39,65 @@ var User = function() {
...
-      else {
-        userConfig.data = {};
-        userConfig.data.apiKey = process.env.MODULUS_TOKEN;
-        userConfig.data.userId = result.id;
-        userConfig.data.username = result.username;
-        callback(null, userConfig);
-      }
+User.prototype.initAuthToken = function (callback) {
+  if (process.env.MODULUS_TOKEN) {
+    this.getForToken(process.env.MODULUS_TOKEN, function (err, result) {
```
Comment (jackboberg) [inline lib/controllers/user.js]: need a `return` here, or bring back the `else`

File: lib/controllers/project.js (hunk 9 above)
Comment (tzmanics) [inline lib/controllers/project.js]: Do you want me to disable the eslint error that is throwing for the un-handled error and make a TODO? or do you have another suggestion?

Comment (jackboberg) [inline lib/controllers/project.js]: Just leave it as is, I don't like leaving it any better and have no other suggestions 😜.

**Comment by tzmanics** [comment]
made the noted changes for this linting step.

**Pull request merged**